tests/ui/literal_string_with_formatting_arg.rs

```diff
@@ -0,0 +1,31 @@
+#![warn(clippy::literal_string_with_formatting_arg)]
+#![allow(clippy::unnecessary_literal_unwrap)]
+
+fn main() {
+    let x: Option<usize> = None;
+    let y = "hello";
+    x.expect("{y} {}"); //~ literal_string_with_formatting_arg
+    x.expect(" {y} bla"); //~ literal_string_with_formatting_arg
+    x.expect("{:?}"); //~ literal_string_with_formatting_arg
+    x.expect("{y:?}"); //~ literal_string_with_formatting_arg
+    x.expect(" {y:?} {y:?} "); //~ literal_string_with_formatting_arg
+    x.expect(" {y:..} {y:?} "); //~ literal_string_with_formatting_arg
+    x.expect(r"{y:?}  {y:?} "); //~ literal_string_with_formatting_arg
+    x.expect(r"{y:?} y:?}"); //~ literal_string_with_formatting_arg
+    x.expect(r##" {y:?} {y:?} "##); //~ literal_string_with_formatting_arg
+    // Ensure that it doesn't try to go in the middle of a unicode character.
+    x.expect("———{:?}"); //~ literal_string_with_formatting_arg
```

Comment:
I feel like this shouldn't be linted, for the simple reason, that the `{:?}` can't refer to anything. If someone mixed `expect()` up to think that it allows formatting, they would probably write `expect("{:?}", obj)` which would result in an error, since expect only takes one argument.

One option to make this line more accepting would be to restrict linting to `{x:?}` / `{x}` where x is available in the current scope :thinking: